src/setfit/trainer.py

```diff
@@ -131,33 +146,70 @@ def _apply_column_mapping(self, dataset: "Dataset", column_mapping: Dict[str, st
         )
         return dataset
 
-    def call_model_init(self):
+    def _hp_search_setup(self, trial: Union["optuna.Trial", Dict[str, Any]]):
+        """HP search setup code"""
```

Comment:
Let's add a note to the source inspiration :)

```suggestion
        """HP search setup code"""
        
         # Heavily inspired by transformers.Trainer._hp_search_setup
         # https://github.com/huggingface/transformers/blob/cbb8a37929c3860210f95c9ec99b8b84b8cf57a1/src/transformers/trainer.py#L1163
```